### PR TITLE
dotCMS/core#19597-reload-content-data-when-dialog-is-closed

### PR DESCRIPTION
**Pull request by alfredo-dotcms**

### CODE REVIEW

File: src/app/view/components/main-legacy/main-legacy.component.ts [hunk 1]
```diff
@@ -11,26 +9,13 @@ import { DotCustomEventHandlerService } from '@services/dot-custom-event-handler
...
 
-    /**
-     * Reload content search iframe when contentlet editor close
-     *
-     * @memberof MainComponentLegacyComponent
-     */
-    onCloseContentletEditor(): void {
-        this.dotIframeService.reloadData(this.dotRouterService.currentPortlet.id);
-    }
-
```
Comment (fmontes) [inline src/app/view/components/main-legacy/main-legacy.component.ts:33]: Why?

**Review by fmontes — COMMENTED** [review]
Can you add in the description what happened here and why?

File: src/app/view/components/main-legacy/main-legacy.component.ts (hunk 1 above)
Comment (alfredo-dotcms) [reply src/app/view/components/main-legacy/main-legacy.component.ts:33]: @fmontes 
Currently when you add a new content in Content --> Search, and when you close the dialog, the code that handles the **Reload of the data** is only present on the `<dot-main-component> (main-legacy.component.ts)` which is the parent component of `<dot-create-contentlet>`

when you put the new Url to create content in the browser, the component that gets loaded is <dot-create-contentlet> so it does not have any parent component, and when the dialog gets closed no data is being reloaded

so I moved the logic that reloads the data from the parent `<dot-main-component> (main-legacy.component.ts)` to the dialog `<dot-create-contentlet>` in that way the data gets reloaded on both scenarios

Comment (fmontes) [reply src/app/view/components/main-legacy/main-legacy.component.ts:33]: Gracias!

**Review by fmontes — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

**Pull request merged**